llvm/lib/Support/VirtualFileSystem.cpp

```diff
@@ -326,7 +326,7 @@ ErrorOr<std::unique_ptr<File>>
 RealFileSystem::openFileForRead(const Twine &Name) {
   SmallString<256> RealName, Storage;
   Expected<file_t> FDOrErr = sys::fs::openNativeFileForRead(
-      adjustPath(Name, Storage), sys::fs::OF_None, &RealName);
+      adjustPath(Name, Storage), sys::fs::OF_Text, &RealName);
```

Comment:
I don't think this is always valid, for eg. `#embed` - We should find a way to propagate that somehow

Reply:
That's a good point, I will look into that, thanks!

Reply:
I have created a separate PR https://github.com/llvm/llvm-project/pull/107906 to propagate an IsText flag to these functions which on z/OS will query the file tag to determine whether to open the file as text or binary.